src/telemetry-data/data-model.md

```diff
@@ -0,0 +1,203 @@
+# Thin-edge data model
+
+## Motivations
+
+* Move telemetry data around thin-edge components: the devices, the cloud, the sensors, the child devices, the analytics engines ...
+* Ease the ingestion of telemetry data coming from various sources, possibly implemented by tiers.
+* Ease the transfer of telemetry data to various cloud.
+* Ease the exchange of telemetry data between processing component running on a thin-edge device.
+
+## Current state of thin-edge JSON version 0.5
+
+Benefits:
+* Simple to learn, yet powerful
+* Requires no configuration
+* Atomic messages can be grouped into larger messages
+  * making the transfers more efficient,
+  * providing a better UX on the cloud console,
+  * and reducing the customer's bill (based on the message rate).
+* Writing a mapper between thin-edge JSON and another format is simple,
+  * as long as there is one-one correspondence between the messages.
+
+Limitations:
+* Many data sources *already* exist and produce data that can be mapped to thin-edge JSON,
+  * but are not *actually* thin-edge JSON,
+  * making mandatory yet another data mapper.
+* Thin-edge JSON is focused on a format, while what we need first is a data model.
+* The current format parser is too rigid,
+  * rejecting the whole message when only one part is not meaningful.
+* Many data sources are publishing data with one topic per kind of data
+  * while thin-edge expect all the data on a single topic.
+* Writing a mapper that groups messages as those produced by `collectd` is a more complex task than expected.
+  * One should leverage the effort done for `collectd` and use it 
+* All the data published on `tedge/measurements` are sent to the cloud.
+  * What if we wish to filter some data out?
+  * What if we wish to pre-process the data to only forward an aggregate?
+* The meaning and purpose of multi-measurements is not clear.
+* Possible name collisions with reserved key words (`time` and `type`)?
+* Non numeric values are rejected.
+* Accept only ISO-8601 timestamp, while Unix Epoch is often used by data sources. 
+* The child device identifiers are expected to be in the topic name,
+  * while some data sources published their identifier along the message.
+* There is no way to attach metadata to a data source (e.g the unit).
+* There is no way to attach metadata to a data point (e.g the quality of the measurement).
```

Comment:
```suggestion
* There is no way to attach metadata to a data point (e.g the quality of the measurement (https://honeywellprocess-community.force.com/opcsupport/s/article/What-are-the-OPC-Quality-Codes)).
```
Just to avoid people confusing quality of measurement with quality of service (unless quality of measurement is a well known term in the IoT world).

Reply:
Added